### PR TITLE
feat(FEC-8046): 360 support

### PR DESCRIPTION
**Pull request by yairans**
### Description of the Changes

* force inBrowserFullscreenForIOS when 'vr' plugin is set
* set `vrStereo` ui component when 'vr' plugin is set
* pass the target dom element to 'vr' plugin

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment 
- [x] Travis tests are passing (or test results are not worse than on master branch :))
- [x] Docs have been updated

### CODE REVIEW

File: src/common/ui-wrapper.js [hunk 1]
```diff
@@ -38,6 +38,14 @@ class UIWrapper {
     this.setConfig(Utils.Object.mergeDeep({}, previewThumbnailConfig, seekbarConfig), 'seekbar');
   }
 
+  setFullscreenConfig(config: ProviderMediaConfigObject): void {
+    if (this._disabled) return;
+    if (Utils.Object.getPropertyPath(config, 'plugins.vr')) {
```
Comment (yairans) [inline src/common/ui-wrapper.js]: maybe better to check the plugin does exists via `getPlugin('vr')`

File: src/kaltura-player.js [hunk 2]
```diff
@@ -77,4 +79,16 @@ export default class KalturaPlayer {
       set: undefined
     };
   }
+
+  _addBindings(): void {
```
Comment (dan-ziv) [inline src/kaltura-player.js]: This is massive and too specific referring between kaltura player and a plugin
If the problem is that the plugin cannot listen to the UI events the solution is to move the plugin manager to the kaltura player level and not bind the kaltura player to a specific plugin

Comment (dan-ziv) [reply src/kaltura-player.js]: one more thing is - if we will run kaltura player in chromeless mode (`{ui: {disable: true}}`) the vr navigation won't work.

File: src/common/plugins/plugins-config.js [hunk 3]
```diff
@@ -12,7 +12,8 @@ function evaluatePluginsConfig(options: KalturaPlayerOptionsObject): void {
   if (options.plugins) {
     const dataModel: Object = {
       pVersion: __VERSION__,
-      pName: __NAME__
+      pName: __NAME__,
+      target: options.targetId
```
Comment (OrenMe) [inline src/common/plugins/plugins-config.js]: target is too generic term, suggest to change to domRootElementId

File: src/common/ui-wrapper.js [hunk 4]
```diff
@@ -38,6 +38,25 @@ class UIWrapper {
...
+    }
+  }
+
+  _setFullscreenConfig(): void {
+    const fullscreenConfig = Utils.Object.getPropertyPath(this._uiManager, 'config.components.fullscreen');
+    this.setConfig(Utils.Object.mergeDeep({}, {inBrowserFullscreenForIOS: true}, fullscreenConfig), 'fullscreen');
+  }
+
+  _setStereoConfig(config: KalturaPlayerOptionsObject): void {
+    if (config.plugins.vr.toggleStereo || (Env.device.type && config.plugins.vr.toggleStereo !== false)) {
```
Comment (OrenMe) [inline src/common/ui-wrapper.js]: not sure I understand this condition

File: src/kaltura-player.js [hunk 5]
```diff
@@ -23,12 +23,14 @@ export default class KalturaPlayer {
     this._logger = getLogger('KalturaPlayer' + Utils.Generator.uniqueId(5));
     this._uiWrapper = new UIWrapper(this._player, options.ui);
     this._provider = new Provider(options.provider, __VERSION__);
+    this._uiWrapper.handleVr(options);
```
Comment (OrenMe) [inline src/kaltura-player.js]: why not handle this in the uiWrapper?

File: src/kaltura-player.js (hunk 2 above)
Comment (OrenMe) [reply src/kaltura-player.js]: @dan-ziv, @yairans  we will move this in to the player and then the UI will call `player.toggleStereoMode()` directly

File: src/common/ui-wrapper.js (hunk 4 above)
Comment (yairans) [reply src/common/ui-wrapper.js]: @OrenMe  `toggleStereo` is `false` by default for desktop and `true` for mobile. is it clear?

File: src/common/plugins/plugins-config.js (hunk 3 above)
Comment (yairans) [reply src/common/plugins/plugins-config.js]: done

File: src/kaltura-player.js (hunk 5 above)
Comment (yairans) [reply src/kaltura-player.js]: done

File: src/kaltura-player.js (hunk 2 above)
Comment (yairans) [reply src/kaltura-player.js]: done

**Pull request merged**